client.go

```diff
@@ -110,6 +114,15 @@ type Client struct {
 	// ResponseLogHook allows a user-supplied function to be called
 	// with the response from each HTTP request executed.
 	ResponseLogHook ResponseLogHook
+
+	// CheckRetry specifies a policy for handling retries. It is called
+	// following each request with the response and error values returned by
+	// the http.Client. If CheckRetry returns false, the Client stops retrying
+	// and returns the response to the caller. If CheckRetry returns an error,
+	// that error value is returned in lieu of the error from the request. If
+	// an error is returned, it is up to the CheckResponse callback to properly
+	// close any response body before returning.
+	CheckRetry func(resp *http.Response, err error) (bool, error)
```

Comment:
I'm not sure I understand, because this will be exposed in the godoc since both Client and CheckRetry are exported. Function type struct fields are standard for callbacks, see Client.CheckRedirect, Transport.Dial, DialTLS, Proxy, Server.ConnState in the http package for example. And I'm not sure what an interface would provide over the ability to use a function literal and closure.